storage/migrations/01_oasis_3_consensus.up.sql

```diff
@@ -251,6 +251,30 @@ CREATE TABLE oasis_3.votes
   PRIMARY KEY (proposal, voter)
 );
 
+-- Related Accounts Data
+
+CREATE TABLE oasis_3.accounts_related_transactions
+(
+  account_address oasis_addr NOT NULL,
+  tx_block UINT63 NOT NULL,
+  tx_index UINT31 NOT NULL,
+  FOREIGN KEY (tx_block, tx_index) REFERENCES oasis_3.transactions(block, tx_index) DEFERRABLE INITIALLY DEFERRED
+);
+CREATE INDEX ix_accounts_related_transactions_address_block_index ON oasis_3.accounts_related_transactions (account_address, tx_block, tx_index);
+
+CREATE TABLE oasis_3.accounts_related_events
```

Comment:
This just duplicates all the fields from `oasis_3.events`, and adds the account_address (probably better renamed to related_account or related_address).
With this table, we don't really need the `events` table (with the minor caveat that we'd have to store a row with `account_address=NULL` for events without related accounts). Or more realistically, we could just add the new column to the old table, and never introduce the new table.

But it's a very inefficient way of storing things. All event data is duplicated for every related account.

I would suggest one of two options (1b and 2 are realistic from what I've thought about it so far):
- **1**: A "classic" SQL approach of representing the many-to-many relationship between events and addresses: A table with just the columns `(event_id, related_account_addr)`. You'd have to add `event_id` to the events table. The potentially tricky part (I haven't thought it through) is that you'd have to insert the event and remember the ID that was assigned to it by the DB, then use that ID when insertnig into the new table. Our `batch` approach isn't easily modified to support that, I think.
- **1b**: Like above, but devise some indexer-internal way of hashing each event (probably based on event type, height, and body) into an ID.
- **1c**: Like above, but use randomly-generated UUIDs for IDs. I mention this option just to dis it :). Randomness would mean different DB data on every re-run, making tests, diffing etc much harder.
- **2**: Expand the `events` table with a column `related_accounts TEXT[]`. We can efficiently query it by creating  a [GIN index](https://www.postgresql.org/docs/9.1/textsearch-indexes.html) over it. The potentially tricky part is inserting the array into the DB using Go's `pgx`; IDK how good the support is, maybe it's easy. 

Sample code for approach 2:
```
indexer=# create table events(id int, rels text[]);                                                                                                                          
CREATE TABLE
indexer=# CREATE INDEX foo_idx ON events USING gin(rels);                             
CREATE INDEX                               
indexer=# insert into events values (1, ARRAY ['foo', 'bar']);            
INSERT 0 1             
indexer=# insert into events values (2, ARRAY ['bar', 'baz']);                                                                                                               
INSERT 0 1           
indexer=# insert into events values (3, ARRAY ['foo', 'qux']);                                                                                                               
INSERT 0 1                                                                                                                                        
indexer=# select * from events where 'bar'=any(rels);                                 
 id |   rels                                                             
----+-----------                                                         
  1 | {foo,bar}                                                          
  2 | {bar,baz}                                                          
(2 rows)                                       
```

Reply:
Thanks! Discussed offline and opted for option 2